worker/src/handlers/aggregationProcessor/aggregationProcessor.js

```diff
@@ -214,12 +237,34 @@ const getAddPipeline = async ({
     };
   }
 
+  // MIDDLE
+
+  let addToMiddle;
+  if (model.lastRun) {
+    addToMiddle = {
+      $and: [
+        {
+          timestamp: {
+            $gte: moment(model.fromTimestamp || now).toDate(),
+            $lt: getAddFromTimestamp({ model, now }).toDate()
+          }
+        },
+        {
+          stored: {
+            $gte: moment(model.lastRun).toDate(),
+          }
+        }
+      ]
```

Comment:
This should be functionally equivalent.
```suggestion
          timestamp: {
            $gte: moment(model.fromTimestamp || now).toDate(),
            $lt: getAddFromTimestamp({ model, now }).toDate()
          },
          stored: {
            $gte: moment(model.lastRun).toDate(),
          }
```